src/core/dfn-panel.js

```diff
@@ -0,0 +1,163 @@
+      return clickedSelfLink ? "HIDE" : "ACTIVATE";
+    }
+    const panel = clickTarget.closest("#dfn-panel");
+    return panel.classList.contains("activated") ? "HIDE" : "NOOP";
+  }
+  if (document.getElementById("dfn-panel")) {
+    return "HIDE";
+  }
+  return "NOOP";
+}
+
+/** @param {HTMLElement} dfn */
+function createPanel(dfn) {
+  const { id } = dfn;
+  const href = `#${id}`;
+
+  const links = [...document.querySelectorAll(`a[href="${href}"]`)];
+
+  /** @type {HTMLElement} */
+  const panel = hyperHTML`
+    <aside class="dfn-panel" id="dfn-panel">
+      <b><a class="self-link" href="${href}">${href}</a></b>
+      ${
+        links.length
+          ? hyperHTML`<b>Referenced in:</b>${referencesToHTML(id, links)}`
+          : null
+      }
+    </aside>
+  `;
+  return document.body.appendChild(panel);
+}
+
+/**
+ * @param {string} id dfn id
+ * @param {HTMLLinkElement[]} links
+ * @returns {HTMLUListElement}
+ */
+function referencesToHTML(id, links) {
+  /** @type {Map<string, string[]>} */
+  const titleToIDs = new Map();
+  links.forEach((link, i) => {
+    const linkID = link.id || `ref-for-${id}-${i + 1}`;
+    if (!link.id) link.id = linkID;
+    const title = getReferenceTitle(link);
+    const ids = titleToIDs.get(title) || titleToIDs.set(title, []).get(title);
+    ids.push(linkID);
+  });
+
+  /** @param {[string, string[]]} entry */
+  const toLinkProps = ([title, ids]) => {
+    return [{ title, id: ids[0] }].concat(
+      ids.slice(1).map((id, i) => ({ title: `(${i + 2})`, id }))
+    );
+  };
+
+  /**
+   * @param {[string, string[]]} entry
+   * @returns {HTMLLIElement}
+   */
+  const listItemToHTML = entry =>
+    hyperHTML`<li>${toLinkProps(entry).map(
+      link => hyperHTML`<a href="#${link.id}">${link.title}</a>${" "}`
+    )}</li>`;
+
+  return hyperHTML`<ul>${[...titleToIDs.entries()].map(listItemToHTML)}</ul>`;
+}
+
+/** @param {HTMLAnchorElement} link */
+function getReferenceTitle(link) {
+  const section = link.closest("section");
+  if (!section) return null;
+  const heading = section.querySelector("h1, h2, h3, h4, h5");
+  if (!heading) return null;
+  return norm(heading.textContent);
+}
+
+/**
+ * @param {HTMLElement} dfn
+ * @param {HTMLElement} panel
+ */
+function displayPanel(dfn, panel) {
+  panel.classList.add("on");
+
+  const dfnRect = dfn.getBoundingClientRect();
+  const panelRect = panel.getBoundingClientRect();
+  const panelWidth = panelRect.right - panelRect.left;
+
+  let top = window.scrollY + dfnRect.top;
+  let left = dfnRect.left + dfnRect.width + 5;
+  if (left + panelWidth > document.body.scrollWidth) {
+    // Reposition, because the panel is overflowing
+    left = dfnRect.left - (panelWidth + 5);
+    if (left < 0) {
+      left = dfnRect.left;
+      top = top + dfnRect.height;
+    }
+  }
+
+  Object.assign(panel.style, { left: `${left}px`, top: `${top}px` });
+}
+
+async function loadStyle() {
```

Comment:
we should make this a util function.

Reply:
As previously noted, doing so will confuse the bundler as it can't track dynamic import expression with dynamic path value. What we need is a way to import CSS as mentioned in [a recent TC39 proposal](https://github.com/tc39/proposal-module-attributes/blob/master/README.md#motivation).